contracts/infrastructure/DappRegistry.sol

```diff
@@ -2,77 +2,197 @@ pragma solidity ^0.6.12;
 
 import "./IAuthoriser.sol";
 import "./dapp/IFilter.sol";
-import "./base/Owned.sol";
 import "./storage/Storage.sol";
 
-contract DappRegistry is IAuthoriser, Storage, Owned {
+contract DappRegistry is IAuthoriser, Storage {
 
-    bytes32 constant internal ARGENT_DEFAULT = keccak256("argent_default");
+    // The timelock period
+    uint64 public securityPeriod;
+    // The new timelock period
+    uint64 public newSecurityPeriod;
+    // Time at which the new timelock becomes effective
+    uint64 public securityPeriodChangeAfter;
 
-    struct Authorisation {
-        bool isActive;
-        address filter;
-    }
+    // bit vector of enabled registry ids for each wallet
+    mapping (address => bytes32) public enabledRegistryIds; // [wallet] => [bit vector of 256 registry ids]
+    // authorised dapps and their filters for each registry id
+    mapping (uint8 => mapping (address => bytes32)) public authorisations; // [registryId] => [dapp] => [{filter:160}{validAfter:64}]
+    // pending authorised dapps and their filters for each registry id
+    mapping (uint8 => mapping (address => bytes32)) public pendingFilterUpdates; // [registryId] => [dapp] => [{filter:160}{validAfter:64}]
+    // owners for each registry id
+    mapping (uint8 => address) public registryOwners; // [registryId] => [owner]
```

Comment:
Since we cannot remove or overwrite registries is it possible that we will run out of these 256 registry ids at some point?

Reply:
No, that shouldn't be an issue since we currently plan to use only 2 registries. Even if we change our minds it will be more than enough.

Reply:
Yes, in theory. In practice we do not expect using more than 2 or 3 registries.